db/transactions.go

```diff
@@ -21,29 +21,35 @@ func NewTxManager(db *sqlx.DB) *TxManager {
 	return &TxManager{DB: db}
 }
 
-func (s *TxManager) WithTransaction(ctx context.Context, f func(context.Context, Querier) error) (err error) {
-	ctx, span := o11y.StartSpan(ctx, "tx-manager: with-transaction")
-	defer o11y.End(span, &err)
+func retryable(err error) bool {
+	return err != nil && errors.Is(err, ErrBadConn)
+}
 
+func (s *TxManager) WithTransaction(ctx context.Context, f func(context.Context, Querier) error) (err error) {
 	// Retry this transaction 3 times
 	for i := 0; i < 3; i++ {
-		err = s.WithOneTransaction(ctx, f)
-		if !errors.Is(err, ErrBadConn) {
+		err = s.WithOneTransaction(ctx, i, f)
+		if !retryable(err) {
 			break
 		}
-		o11y.AddField(ctx, "bad_con", i)
-		o11y.AddField(ctx, "warning", err)
 	}
 
 	// Note that the above defer can reassign err
 	return err
 }
 
-func (s *TxManager) WithOneTransaction(ctx context.Context, f func(context.Context, Querier) error) (err error) {
+func (s *TxManager) WithOneTransaction(ctx context.Context,
```

Comment:
I wonder if `WithOneTransaction` should be a locally scoped func? So we can share the magic number 3, and makes the warning -> error on failure logic easier?

Reply:
technically a breaking change, but i don't see any use of `WithOneTransaction` in the circleci org, so seems like a good idea

Reply:
I tried it and didn't seem like an improvement, so i'm keeping what we have now